### ISSUE TITLE
Add option to remove "broken" links

### ISSUE THREAD

**Issue by Werzi2001** [mkdocs/mkdocs#1443]
Currently a broken link just causes a warning/error (depending on strict configuration) but the broken link stays in the documentation. I'd like to have an option to remove broken links instead.

In my case the documentation is built dynamically and depending on structure linked pages might be there or not. This is not a problem but the incorrect links have to be removed of course in that case.

**Comment by waylan** [comment]
I believe the assumption is that broken links exist due to typographic errors. In other words, the user sees the warning/error and goes back to edit the page and fix the link before building again. I don't think we ever anticipated that the user would ignore the warning and leave things as-is, which seems to be the presumption of this request. I'll have to think about this.

**Comment by Werzi2001** [comment]
Yes exactly. Our documentation is split into multiple parts which can be combined in any way. Links between the parts might be broken if the target part is not included (which is expected to happen).

Of course I wouldn't want to change standard behaviour. Just add a new configuration like `remove_broken_links`.

**Comment by waylan** [comment]
We generally try to avoid adding more config settings. If anything, we look for ways to reduce the number of settings. In fact, this seems like the sort of thing that would be best implemented by a third party plugin or Markdown extension. In fact, the code which issues the existing warning is implemented as a custom Markdown extension which is included as part of MkDocs.

**Comment by pawamoy** [comment]
The initial feature request is a bit ambiguous: what does "removing a broken link" mean? Do you want to remove `<a>` tags around hyperlinks? Do you want to remove the `<a>` and their inner text entirely? Do you have broken links only in Markdown pages, or in the navigation too? What should happen with those? Is it only about internal links, or external ones too? Only about pages, or assets too?

Since version 1.5, [MkDocs is able to validate several things](https://www.mkdocs.org/user-guide/configuration/#validation). Would these validation mechanisms work for you? You can tune down specific warnings so they don't fail your build.

If you really want to remove links (whatever this means), the way to go is what @waylan described: using a custom Markdown extension (more specifically, a tree processor), see for example https://github.com/mkdocs/mkdocs/blob/59a295f5f9cbd07dcb52c70d53d16523f24e2919/mkdocs/structure/pages.py#L345.

In the meantime I'll close as out of scope for MkDocs core, but feel free to comment further.
